src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2692,3 +2725,319 @@ map<long, long> JasmineGraphInstanceService::calculateLocalOutDegreeDistribution
+    std::string centralStoreFile = aggregatorFilePath + "/" + centralGraphIdentifier;
+    instance_logger.log("###INSTANCE### centralstore " + centralStoreFile,"info");
+
+    for (vector<string>::iterator workerIt=workerSockets.begin(); workerIt!=workerSockets.end(); ++workerIt) {
+
+        std::vector <string> workerSocketPair;
+        boost::split(workerSocketPair, *workerIt, boost::is_any_of(":"));
+        struct stat s;
+        if (stat(centralStoreFile.c_str(),&s) == 0) {
+            if (s.st_mode & S_IFREG) {
+
+                string host = workerSocketPair[0];
+                int port = stoi(workerSocketPair[1]);
+                int workerGraphID = stoi(workerSocketPair[2]);
+                int dataPort = stoi(workerSocketPair[3]);
+
+                instance_logger.log("thisWorkerPort " + to_string(thisWorkerPort) + " port " + to_string(port), "info");
+
+                if (port == thisWorkerPort) {
+                    continue;
+                }
+
+                Utils utils;
+                bool result = true;
+                std::cout << pthread_self() << " host : " << host << " port : " << port << " DPort : " << dataPort << std::endl;
+                int sockfd;
+                char data[300];
+                bool loop = false;
+                socklen_t len;
+                struct sockaddr_in serv_addr;
+                struct hostent *server;
+
+                sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+                if (sockfd < 0) {
+                    instance_logger.log( "Cannot accept connection" , "error");
+                    return 0;
+                }
+
+                if (host.find('@') != std::string::npos) {
+                    host = utils.split(host, '@')[1];
+                }
+
+                server = gethostbyname(host.c_str());
+                if (server == NULL) {
+                    instance_logger.log("ERROR, no host named " , "error");;
+                }
+
+                bzero((char *) &serv_addr, sizeof(serv_addr));
+                serv_addr.sin_family = AF_INET;
+                bcopy((char *) server->h_addr,
+                      (char *) &serv_addr.sin_addr.s_addr,
+                      server->h_length);
+                serv_addr.sin_port = htons(port);
+                if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+                    instance_logger.log("ERROR connecting", "error");
+                    //TODO::exit
+                }
+
+                bzero(data, 301);
+                int result_wr = write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
+
+                if(result_wr < 0) {
+                    instance_logger.log("Error writing to socket", "error");
+                }
+
+                instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
+                bzero(data, 301);
+                read(sockfd, data, 300);
```

Comment:
Can we have a constant instead of 300 and 301?

Reply:
Added a constant

Reply:
Ok